system/lib/libcxxabi/src/cxa_exception_emscripten.cpp

```diff
@@ -130,3 +130,21 @@ char* __get_exception_terminate_message(void* thrown_object) {
 }
 
 #endif // __USING_EMSCRIPTEN_EXCEPTIONS__ || __USING_WASM_EXCEPTIONS__
+
+#ifndef __USING_WASM_EXCEPTIONS__
+
+namespace __cxxabiv1 {
+
+void* __cxa_allocate_exception(size_t size) _NOEXCEPT {
+  // Thrown object is prepended by exception metadata block
+  __cxa_exception* ex = (__cxa_exception*)malloc(size + sizeof(__cxa_exception));
+  return thrown_object_from_cxa_exception(ex);
+}
+
+void __cxa_free_exception(void *thrown_object) _NOEXCEPT {
+  free(cxa_exception_from_thrown_object(thrown_object));
+}
```

Comment:
Can we share these functions with the existing functions in `cxa_exceptions.cpp`?
https://github.com/emscripten-core/emscripten/blob/4681c920d0de95533f9728e2acf3144b6998d048/system/lib/libcxxabi/src/cxa_exception.cpp#L178-L207

If not, why?

Reply:
cxa_exception.cpp is not compiled at all when building with emscripten exceptions.    We could look at compiling part of it in this mode, but I think that should be left for a followup.  

I think the idea is that `cxa_exception_emscripten.cpp` is designed to be compiled instead of `cxa_exception.cpp`

Reply:
> I think the idea is that `cxa_exception_emscripten.cpp` is designed to be compiled instead of `cxa_exception.cpp`

In Wasm EH we compile both, because `cxa_exception_emscripten.cpp` contains exception printing functions. I thought the idea was we put Emscripten (including Wasm) specific functions.. Anyway, I guess we can't just add `cxa_exceptions.cpp` in Emscripten EH bc there are many conflicting functions, so doing this separate LGTM atm.